src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexParser.cs

```diff
@@ -0,0 +1,1916 @@
+            }
+
+            var afterQuestionPos = _lexer.Position;
+
+            // Lots of possible options when we see (?.  Look at the immediately following character
+            // (without any allowed spaces) to decide what to parse out next.
+            ConsumeCurrentToken(allowTrivia: false);
+            switch (_currentToken.Kind)
+            {
+                case RegexKind.LessThanToken:
+                    // (?<=...) or (?<!...) or (?<...>...) or (?<...-...>...)
+                    return ParseLookbehindOrNamedCaptureOrBalancingGrouping(openParenToken, questionToken);
+
+                case RegexKind.SingleQuoteToken:
+                    //  (?'...'...) or (?'...-...'...)
+                    return ParseNamedCaptureOrBalancingGrouping(
+                        openParenToken, questionToken, _currentToken);
+
+                case RegexKind.OpenParenToken:
+                    // alternation construct (?(...) | )
+                    return ParseConditionalGrouping(openParenToken, questionToken);
+
+                case RegexKind.ColonToken:
+                    return ParseNonCapturingGroupingNode(openParenToken, questionToken);
+
+                case RegexKind.EqualsToken:
+                    return ParsePositiveLookaheadGrouping(openParenToken, questionToken);
+
+                case RegexKind.ExclamationToken:
+                    return ParseNegativeLookaheadGrouping(openParenToken, questionToken);
+
+                case RegexKind.GreaterThanToken:
+                    return ParseNonBacktrackingGrouping(openParenToken, questionToken);
+
+                default:
+                    if (_currentToken.Kind != RegexKind.CloseParenToken)
+                    {
+                        // Native parser reports "Unrecognized grouping construct", *except* for (?)
+                        openParenToken = openParenToken.AddDiagnosticIfNone(new EmbeddedDiagnostic(
+                            WorkspacesResources.Unrecognized_grouping_construct,
+                            openParenToken.GetSpan()));
+                    }
+
+                    break;
+            }
+
+            // (?)
+            // Parse this as a normal group. The question will immediately error as it's a 
+            // quantifier not following anything.
+            _lexer.Position = afterQuestionPos - 1;
+            return ParseSimpleGroup(openParenToken);
+        }
+
+        private RegexConditionalGroupingNode ParseConditionalGrouping(RegexToken openParenToken, RegexToken questionToken)
+        {
+            var innerOpenParenToken = _currentToken;
+            var afterInnerOpenParen = _lexer.Position;
+
+            var captureToken = _lexer.TryScanNumberOrCaptureName();
+            if (captureToken == null)
+            {
+                return ParseConditionalExpressionGrouping(openParenToken, questionToken, innerOpenParenToken);
+            }
+            else
```

Comment:
Could remove the `else` since bailed out earlier.

Reply:
#resolved.